actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -6,15 +6,34 @@ def send_action(method, *args)
       ret
     end
 
+    # Perform the default rendering for the given action and variant.
+    # Essentially calls +render+ with +args+ if there is an appropriate template for the request.
+    #
+    # If there is no appropriate template, and no block is given, defers to
+    # +default_missing_template_action+.  Otherwise, executes the block.
+    #
+    # +args+:: passed through to render
     def default_render(*args)
       if template_exists?(action_name.to_s, _prefixes, variants: request.variant)
         render(*args)
       else
-        logger.info "No template found for #{self.class.name}\##{action_name}, rendering head :no_content" if logger
-        head :no_content
+        if block_given?
+          yield(*args)
+        else
+          default_missing_template_action(*args)
```

Comment:
I'm not sure we should pass the args here (I know `default_render` does it, but still). What do you think, @rafaelfranca?